### PR TITLE
Add additional Jira columns

### PR DESCRIPTION
**Pull request by BENBEN5959**

### CODE REVIEW

File: src/proxy-page/steps/addJira.ts [hunk 1]
```diff
@@ -211,135 +238,178 @@ export default (config: ConfigService, jiraService: JiraService): Step => async
...
+        ?? config.get('confluence.baseURL');
       issues.forEach((issue) => {
         data.push({
-          key: {
-            name: issue.key,
-            link: `${baseUrl}/browse/${issue.key}?src=confmacro`,
-          },
+          reporter: issue.fields.reporter?.displayName || '',
+          components: getContentWithMap(issue.fields.components),
+          acceptance_criteria: getTextContent(issue.fields.customfield_10042),
```
Comment (PiiXiieeS) [inline src/proxy-page/steps/addJira.ts]: This `customfield_10042` looks like a custom field for us.
What would happen if we run this code in another Jira/Confluence instance?

File: src/proxy-page/steps/addJira.ts [hunk 2]
```diff
@@ -211,135 +238,178 @@ export default (config: ConfigService, jiraService: JiraService): Step => async
...
+          detail_design_reference: getTextContent(issue.fields.customfield_10129),
```
Comment (PiiXiieeS) [inline src/proxy-page/steps/addJira.ts]: This `customfield_10129` looks like a custom field for us.
What would happen if we run this code in another Jira/Confluence instance?

File: src/proxy-page/steps/addJira.ts [hunk 3]
```diff
@@ -211,135 +238,178 @@ export default (config: ConfigService, jiraService: JiraService): Step => async
...
+          storypoints: issue.fields.customfield_10026 || '',
```
Comment (PiiXiieeS) [inline src/proxy-page/steps/addJira.ts]: This `customfield_10026` looks like a custom field for us.
What would happen if we run this code in another Jira/Confluence instance?

File: src/proxy-page/steps/addJira.ts [hunk 4]
```diff
@@ -211,135 +238,178 @@ export default (config: ConfigService, jiraService: JiraService): Step => async
...
+          labels: issue.fields.labels || [],
+          sprint: getContentWithMap(issue.fields.customfield_10020),
```
Comment (PiiXiieeS) [inline src/proxy-page/steps/addJira.ts]: This `customfield_10020` looks like a custom field for us.
What would happen if we run this code in another Jira/Confluence instance?

File: src/proxy-page/steps/addJira.ts [hunk 5]
```diff
@@ -211,135 +238,178 @@ export default (config: ConfigService, jiraService: JiraService): Step => async
...
-              month: 'short',
-              day: '2-digit',
-              hour: '2-digit',
-              minute: '2-digit',
-            })}`
-            : '',
-          assignee: issue.fields.assignee?.displayName,
+          summary: createLinkObject(issue.key, baseUrl, issue.fields.summary),
+          updated: formatDate(issue.fields.updated),
+          startdate: formatDate(issue.fields.customfield_10015),
```
Comment (PiiXiieeS) [inline src/proxy-page/steps/addJira.ts]: This `customfield_10015` looks like a custom field for us.
What would happen if we run this code in another Jira/Confluence instance?

**Review by PiiXiieeS — COMMENTED** [review]
@BENBEN5959 and @krystiankedra 
I made a review because those fields look like custom fields special for our environment and probably this would fail if deployed to be run against another Jira/Confluence instance.
* `customfield_10015`
* `customfield_10020`
* `customfield_10026`
*  `customfield_10129`
*  `customfield_10042`

Can you check how this work in another instance?

File: src/proxy-page/steps/addJira.ts [hunk 6]
```diff
@@ -170,176 +171,179 @@ export default (config: ConfigService, jiraService: JiraService): Step => async
...
+      };
+      const createColumns = (data) => `[${data.slice(0, 1).flatMap((obj) =>
+        Object.keys(obj)
+          .map((key) => {
+            const field = obj[key];
+            const { gridtype } = field;
+            const { name } = field;
+            return columnConfig[gridtype](name);
+          })
+          .filter((newColumn) => newColumn !== undefined)).join(',')}]`;
```
Comment (krystiankedra) [inline src/proxy-page/steps/addJira.ts]: .filter((newColumn) => newColumn !== undefined)) -> .filter(Boolean)

**Review by krystiankedra — APPROVED** [review]

**Review by krystiankedra — APPROVED** [review]

**Pull request merged**